src/main/java/org/jabref/gui/maintable/MainTableColumnModel.java

```diff
@@ -0,0 +1,125 @@
+package org.jabref.gui.maintable;
+
+import java.util.Objects;
+
+import org.jabref.gui.util.FieldsUtil;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.field.FieldFactory;
+
+public class MainTableColumnModel {
+
+    public enum Type {
+        EXTRAFILE("extrafile", Localization.lang("File type")),
+        FILES("files", Localization.lang("Linked files")),
+        GROUPS("groups", Localization.lang("Groups")),
+        LINKED_IDENTIFIER("linked_id", Localization.lang("Linked identifiers")),
+        NORMALFIELD("field"),
+        SPECIALFIELD("special", Localization.lang("Special"));
+
+        private String name;
+        private String displayName;
+
+        Type(String name) {
+            this.name = name;
+            this.displayName = name;
+        }
+
+        Type(String name, String displayName) {
+            this.name = name;
+            this.displayName = displayName;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public String getDisplayName() {
+            return displayName;
+        }
+
+        public static Type parse(String text) {
+            switch (text) {
+                case "groups": return GROUPS;
```

Comment:
You can use Enum.valueOf() to parse enum keys stored in strings (see for example the group mode union/intersection) thing.

Reply:
I will look into this. A very first try did not work. I probably have to put the enum in an extra file.

Reply:
You have to call `.name() `on the enum keys for storing. 
like here:
```
    public void setGroupViewMode(GroupViewMode mode) {
        put(GROUP_INTERSECT_UNION_VIEW_MODE, mode.name());
    }
```

Reply:
valueOf did not really work, as it parses only the name of the constant, but the text is different from the constant name ("linked_id" instead of "LINKED_IDENTIFIER" and "field" instead of "NORMALFIELD"). But I found another solution (with a little inspiration of stackoverflow) which should be somewhat generic approach.